src/frontend/src/lib/canisters/reward.canister.ts

```diff
@@ -0,0 +1,50 @@
+import type {
+	ClaimVipRewardResponse,
+	NewVipRewardResponse,
+	_SERVICE as RewardService,
+	UserData,
+	VipReward
+} from '$declarations/rewards/rewards.did';
+import { idlFactory as idlCertifiedFactoryReward } from '$declarations/rewards/rewards.factory.certified.did';
+import { idlFactory as idlFactoryReward } from '$declarations/rewards/rewards.factory.did';
+import { getAgent } from '$lib/actors/agents.ic';
+import type { CreateCanisterOptions } from '$lib/types/canister';
+import { Canister, createServices } from '@dfinity/utils';
+
+export class RewardCanister extends Canister<RewardService> {
+	static async create({
+		identity,
+		...options
+	}: CreateCanisterOptions<RewardService>): Promise<RewardCanister> {
+		const agent = await getAgent({ identity });
+
+		const { service, certifiedService, canisterId } = createServices<RewardService>({
+			options: {
+				...options,
+				agent
+			},
+			idlFactory: idlFactoryReward,
+			certifiedIdlFactory: idlCertifiedFactoryReward
+		});
+
+		return new RewardCanister(canisterId, service, certifiedService);
+	}
+
+	getUserInfo = (): Promise<UserData> => {
+		const { user_info } = this.caller({ certified: true });
```

Comment:
Is this correct, `getUserInfo` will always be requested with an update call?

Reply:
@peterpeterparker I guess I do not understand exactly what you mean by this.

Reply:
Are you planning to always call those getters using update calls (certified), or will you also use query calls (not certified)?

Reply:
To be honest this was not a conscious decision. How can i find out what the best solution would be?

Reply:
Take a look at the DID file to determine if it's technically feasible and supported (e.g., `['query']` at the end of a function's declaration). If it is supported, consider whether you need to display the information to the end user as quickly as possible. If the answer is yes, you likely want to support it. Does that help?